### DISCUSSION TITLE
Poetry +  Tox install local packages or git packages

### DISCUSSION THREAD

**Discussion by incoggnito** [tox-dev/tox#1923] [Q&A]
This is my current tox config:

```ini
[testenv]
commands =
    poetry install -v
    poetry run pytest --cov
deps = poetry
description = Test Python code.

[testenv:lint]
commands =
    poetry install -v
    poetry run bandit -ilr src
    poetry run black --check .
    poetry run python scripts/docs.py build
    poetry run flakehell scripts src tests
    poetry run mypy scripts src tests
deps = poetry
description = Run static analyzers on Python code.

[tox]
envlist = lint, py37, py38, py39
isolated_build = true

[tox:.package]
basepython = python3.8
```

The individual commands work in my virtual environment with poetry.
But i use a local package in my development, that is not deployed to pypi.

When i run tox it does not find the local package...
Should I install the local package from my githost? Is this even possible?
I guess for the gitlab-ci worflow i will need this too...

Thanks for any advice!

**Comment by gaborbernat** [comment]
You could use commands_pre to install local packages via pip before commands are run 🤔 If these are additional packages besides the one under tests.

**Reply by incoggnito** [reply]
Ok, you would not include them with poetry?

**Reply by gaborbernat** [reply]
Wait, then I don't understand your question. Can you detail better what you have and what you want to achieve? How you're using tox :-)

**Reply by incoggnito** [reply]
I'm new to tox and poetry, that's why i may ask the wrong way. As far as I understand, I only use dependencies that are described in the poetry-lock-file. Now I have installed a local package with a relative path via poetry. This works so far. But that doesn't seem to work for tox. I guess tox uses pip to install the packages. Maybe I can replace the local package with a git package within poetry, then it should also be able to installed by tox.

**Reply by gaborbernat** [reply]
> Now I have installed a local package with a relative path via poetry.

Probably don't do this 🤔 Specify that local dependency within your poetry lock file instead.
